gdx/src/com/badlogic/gdx/utils/SortedIntList.java

```diff
@@ -123,6 +123,10 @@ public class Iterator implements java.util.Iterator<Node<E>> {
 		private Node<E> position;
 		private Node<E> previousPosition;
 
+		public Iterator () {
+			reset();
+		}
```

Comment:
Ah, yes, this is cleaner than before. There is something to note with this: `iterator()` now calls `reset()` twice if `allocateIterators` is false and the cached Iterator is null. I don't think it's a major issue, but it's a very easy fix in `iterator()` to just `return new Iterator();` in the second if block instead of creating one and then resetting it again.

Reply:
no problemo